src/components/Home.js

```diff
@@ -2,16 +2,24 @@ import React from 'react';
 import getToken from '../lib/tokens.js';
 import { useHistory } from 'react-router-dom';
 
-const Home = () => {
+const Home = (props) => {
   const history = useHistory();
 
   const generateToken = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+    props.setAuth(true);
```

Comment:
We're setting the App component's `auth` state to `true` here, whereas it's set to `localStorage.getItem('token')` in the App component itself on initialization.
For the sake of consistency I think it'd be better to not have the variable capable of multiple types (string, boolean, integer, etc.). Though the variable value is somewhat arbitrary due to us only checking that it's **_truthy_** in the App component, we could just go ahead and set it similarly to the token string:
```suggestion
    props.setAuth(localStorage.getItem('token'));
```